### PR TITLE
Feat: Bookmark post

### PR DESCRIPTION
**Pull request by bharati-21**
This PR contains the following -
Users can bookmark a post from the home page.
Users can view the bookmarked posts from the bookmarks page.
Edited the UserController to add the post id instead of the reference to the post.

### CODE REVIEW

File: src/routes/ProtectedRoutes.jsx [hunk 1]
```diff
@@ -3,13 +3,18 @@ import { Outlet, useLocation, Navigate } from "react-router-dom";
...
 	const { isAuth } = useSelector(getAuthState);
 	const location = useLocation();
 
 	return isAuth ? (
-		<Outlet />
+		<main className="min-h-[100vh] w-full px-8 sm:px-6 grid grid-cols-1 md:grid-container">
+			<LeftSidebar />
+			<Outlet />
+			<RightSidebar />
+		</main>
```
Comment (rohanmathur91) [inline src/routes/ProtectedRoutes.jsx:17]: Liked the implementation of the layout.

**Review by rohanmathur91 — APPROVED** [review]
- Code is very well written
- Liked the folder structure
- Looks good to merge.

**Comment by poojakelkar** [comment]
1. code looks clean
2. proper naming
3. good UI

File: src/features/Post/components/PostItem/index.jsx [hunk 2]
```diff
@@ -45,12 +52,17 @@ const PostItem = ({ post }) => {
 		getIsPostLikedByAuthUser()
 	);
 
+	const getIsPostInBookmarks = () => bookmarks.includes(_id);
+
+	const [isPostInBookmarks, setIsPostInBookmarks] = useState(
+		getIsPostInBookmarks()
```
Comment (ravikumar0403) [inline src/features/Post/components/PostItem/index.jsx]: check this out and [Avoid recreating the initial state](https://beta.reactjs.org/apis/usestate#avoiding-recreating-the-initial-state)

**Review by ravikumar0403 — APPROVED** [review]
- Adding a new post shows a loader for all the posts, can improve that to show a loader without removing all the posts
- Cannot navigate to the previous page after login i.e from GitHub -> login page -> home page. Now if I press the back button, it keeps redirecting me to the home page. 
Let me go back to GitHub, I want to complete the review :(

Check out these issues and one comment, Apart from these everything looks good

**Comment by bharati-21** [comment]
> * Adding a new post shows a loader for all the posts, can improve that to show a loader without removing all the posts

Thank you for your review.

**Comment by bharati-21** [comment]
> 1. code looks clean
> 2. proper naming
> 3. good UI

Thank you for your review.

**Comment by bharati-21** [comment]
> 1. code looks clean
> 2. proper naming
> 3. good UI

Thank you for review!

File: src/features/Post/components/PostItem/index.jsx (hunk 2 above)
Comment (bharati-21) [reply src/features/Post/components/PostItem/index.jsx]: Yes, I have made this change 👍🏽 
Thanks!

**Comment by bharati-21** [comment]
> * Adding a new post shows a loader for all the posts, can improve that to show a loader without removing all the posts
> * Cannot navigate to the previous page after login i.e from GitHub -> login page -> home page. Now if I press the back button, it keeps redirecting me to the home page.
>   Let me go back to GitHub, I want to complete the review :(
> 
> Check out these issues and one comment, Apart from these everything looks good

- I shall hanlde removing loading posts loader once I'm done with the basic requirements.
- Changed the navigation. Thank you for pointing it out.
- Thank you for your review!

**Pull request merged**